GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,478 @@
+
+
+        #region Transformation
+
+        /// <summary>
+        /// Moves the Region in the indicated direction.
+        /// </summary>
+        /// <param name="x">The X-value by which to shift the region</param>
+        /// <param name="y">The Y-value by which to shift the region</param>
+        /// <returns></returns>
+        public virtual Region Translate(int x, int y)
+            => Translate((x, y));
+
         /// <summary>
-        /// Removes (from this region) the outer points shared in common
-        /// with the imposing region
+        /// Moves the region in the indicated direction.
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingPoints(Region imposing)
+        /// <param name="by">The amount (X and Y) to translate this region by.</param>
+        /// <returns>A new, translated region</returns>
+        public virtual Region Translate(Point by)
         {
-            foreach (Point c in imposing.Points)
-            {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
-            }
+            var nw = NorthWestCorner + by;
+            var ne = NorthEastCorner + by;
+            var se = SouthEastCorner + by;
+            var sw = SouthWestCorner + by;
+            return new Region(nw, ne, se, sw);
         }
-
         /// <summary>
         /// Rotates a region around it's center.
         /// </summary>
         /// <param name="degrees">The amount of degrees to rotate this region</param>
-        /// <returns>A region equal to the original region rotated by the given degree</returns>
-        public virtual Region Rotate(double degrees) => Rotate(degrees, Center);
+        /// <returns>A region, equal to the original region rotated by the given degree</returns>
+        public virtual Region Rotate(double degrees)
+            => Rotate(degrees, Center);
 
         /// <summary>
         /// Rotates this region by an arbitrary number of degrees
         /// </summary>
         /// <param name="degrees">The amount of degrees to rotate this region</param>
         /// <param name="origin">The Point around which to rotate</param>
-        /// <returns>This region, rotated.</returns>
-        /// <remarks>
-        /// This is destructive to the region's <see cref="Connections"/>, so you should try to refrain from generating
-        /// those until after you've performed your rotations.
-        /// </remarks>
+        /// <returns>A region, equal to the original region rotated by the given degree</returns>
         public virtual Region Rotate(double degrees, Point origin)
         {
-            degrees = MathHelpers.WrapAround(degrees, 360);
+            degrees %= 360;
```

Comment:
I believe this would fail to account for negative degrees properly because modulo will _not_ wrap -1 to 359:
```suggestion
            degrees = MathHelpers.WrapAround(degrees, 360);
```